pkg/util/goroutinemap/common/exponential_backoff.go

```diff
@@ -0,0 +1,77 @@
+/*
```

Comment:
This logic is pretty specific to the way that goroutinemap operates. The package clayton mentioned, `pkg/util/wait/ExponentialBackoff`, is a more generic mechanism for doing exponential backoff and is already in `pkg/util`. That package takes responsibility for execution (repeatedly execute function with exp backoff until it succeeds). Whereas goroutinemap expects the caller to control function execution and simply blocks it if it is too frequent. Combining the two is a non-trivial task and outside the scope of this PR. And because this backoff mechanism is unique to the way that goroutinemap operates I'd prefer to leave it here until a more through refactor can be done combining the two.